### ISSUE TITLE
Move Python unit tests to pytest

### ISSUE THREAD

**Issue by paddatrapper** [LibreTime/libretime#1178]
Our Python test suite is executed by nose. During the meet up on 17 Feb 2021, we decided to move them to [pytest](https://docs.pytest.org/en/stable/index.html). This should help with #1125 and better reporting of errors.

**Comment by paddatrapper** [comment]
Blocked by #1219

**Comment by jooola** [comment]
@paddatrapper I migrated the analyzer and the api_clients to pytest, do we also want to migrate the django tests to pytest ? Am I missing something else ?

**Comment by paddatrapper** [comment]
Yeah, moving the API tests would be good. I don't think the rest have tests
